### PR TITLE
Add Terraform code to deploy the CCF application on AWS

### PR DESCRIPTION
**Pull request by dfratte**
## Description of Change

This PR brings in a deployment option for AWS cloud using Terraform. It consists on a basic setup using EC2, EIP, Route53, and IAM. You can read more about it the README hosted on the same `terraform` directory that is hosting the infrastructure code.

## Checklist

- [x] PR description included and stakeholders cc'd
- [x] relevant documentation is changed or added

## Notes

N/A

© 2021 Thoughtworks, Inc.

### CODE REVIEW

**Comment by dfratte** [comment]
@ccasher and @dtoakley this is a "generalized" and "parametrizable" version of the terraform code we've been using recently on a client engagement. All feedback is welcome.

File: terraform/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# CCF Infrastructure Setup (AWS/Terraform)
+
+## Overview
+
+This directory contains basic infrastructure code to setup the application in AWS using basic services such as EC2, IAM, Route53. The goal is to have a simple and generic setup that can be configured to each customer needs.
+
+Although the aim was to remain generic, some things might not be entirely applicable to your particular case e.g. you might not have a VPN setup to access private compute instances. You should consider revisit the architecture and adjust the infrastucture accordingly.
```
Comment (nahdelgado) [inline terraform/README.md]:
```suggestion
Although the aim was to remain generic, some things might not be entirely applicable to your particular case e.g. you might not have a VPN setup to access private compute instances. You should consider revisiting the architecture and adjusting the infrastructure accordingly.
```

File: terraform/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+## Architecture
+
+This general architecture for the infrastucture components consist of the following components:
```
Comment (nahdelgado) [inline terraform/README.md]:
```suggestion
The general architecture consists of the following components:
```

File: terraform/README.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+- IAM Policies and Roles
+- ElasticIP
+- Route53 DNS Record
+- S3 Bucket (Terraform State Management)
+
+### EC2 instance
+
+It is the compute resource on which the CCF tool will run. It is configured to run on a private subnet, and to be provisioned using user data.
+
+The user data script is provided in the `install.sh` file, and it contains all the necessary steps to setup the application in this cloud instance. Make sure that you replace EVERY placeholder within that install file with the appropriate values to your case. This are exactly the same ones that you configure the `.env` files for both the client and the API. [Read more](https://www.cloudcarbonfootprint.org/docs/aws).
```
Comment (nahdelgado) [inline terraform/README.md]:
```suggestion
The user data script is provided in the `install.sh` file, and it contains all the necessary steps to setup the application in this cloud instance. Make sure that you replace EVERY placeholder within that install file with the appropriate values to your case. These are exactly the same ones that you would use to configure the `.env` files for both the client and the API. [Read more](https://www.cloudcarbonfootprint.org/docs/aws).
```

File: terraform/README.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+### Elastic IP
+
+We create an ElasticIP to be attached to the EC2 instance. This EIP will later be associated to the DNS record.
+
+### Route53 DNS record
+
+A DNS record within a private/public hosted zone that will allow to map a `HOST` name to the EIP private IP (your instance IP).
+
+### S3 buck`et
```
Comment (nahdelgado) [inline terraform/README.md]:
```suggestion
### S3 bucket
```

**Review by nahdelgado — COMMENTED** [review]
Just minor wording comments.

File: terraform/README.md (hunk 1 above)
Comment (ccasher) [inline terraform/README.md]: revisiting

**Review by ccasher — COMMENTED** [review]
This looks great! Thank you very much for adding this. One small suggestion would be to move these changes into a subdirectory within `terraform/` called `aws/` since this is very specific to AWS.

Also, just as Dan mentioned in the [IAM Auth Mode PR](https://github.com/cloud-carbon-footprint/cloud-carbon-footprint/pull/564) review, it would be good to include documentation on this in `microsite/docs/AWS.md` as well

**Comment by dfratte** [comment]
Last commit should cover latest feedback (specific aws dir, and modify the microsite docs) 👌

**Comment by ccasher** [comment]
Nice work!

**Pull request merged**